python/mdvtools/conversions.py

```
@@ -1,68 +1,250 @@
+            matrix  = mdata.layers[layer]
+            matrix,sparse = get_matrix(matrix,mdata.obs_names,md.obs_names)
+            p.add_rows_as_columns_subgroup("cells",mod,f"{mod}_{layer}",matrix,sparse=sparse, chunk_data=chunk_data)
     return p
 
 
+# The main_names correspond to obs_names in the main anndata object
+# and the mod_names those in the modality.
+# Only required if data is being added from a modality,as sometimes
+# the modality's obs_names will be in a different order and/or a subset of the main names
+# Hence a sparse matrix corresponding to the main indices needs to be created
+def get_matrix(matrix,main_names=[],mod_names=[]):
```

Ruff (0.11.9)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)